### PR TITLE
chore: export openapi schema

### LINKED ISSUE #3420: [datasets] export openapi schema

### PR DESCRIPTION
**Pull request by axiomofjoy**
- Exports OpenAPI schema as a YAML file inside new `schemas` directory.
- Adds a `hatch` command to accomplish this.
- Adds CI to ensure that the exported schema is up to date with the code.

resolves #3420

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,405 @@
+info:
```
Comment (mikeldking) [inline openapi.yaml:1]: thoughts on putting this in a /schemas dir? Root dir works too. Just might get cluttered

**Review by mikeldking — APPROVED** [review]
dope!

File: openapi.yaml (hunk 1 above)
Comment (mikeldking) [reply openapi.yaml:1]: just cuz I want to put the gql schema there too...

Comment (axiomofjoy) [reply openapi.yaml]: Addressed in [1773247](https://github.com/Arize-ai/phoenix/pull/3419/commits/177324745d0849c5c25c85e37d5dd622b9345e7b)

**Pull request merged**